nebula_decoders/include/nebula_decoders/nebula_decoders_hesai/decoders/hesai_decoder.hpp

```diff
@@ -39,48 +41,92 @@ namespace nebula::drivers
+  uint16_t mutliple_return_point_count = 0;
+  uint16_t total_kept_point_count = 0;
+  uint16_t invalid_packet_count = 0;
+  float cloud_distance_min_m = 0;
+  float cloud_distance_max_m = 0;
+  float cloud_azimuth_min_rad = 0;
+  float cloud_azimuth_max_rad = 0;
+  uint64_t packet_timestamp_min_ns = 0;
+  uint64_t packet_timestamp_max_ns = 0;
 
   void clear()
   {
-    distance_counter = 0;
-    fov_counter = 0;
-    raw_azimuth_start = 0;
-    raw_azimuth_end = 0;
-    packet_timestamp_start = 0;
-    packet_timestamp_end = 0;
-    point_azimuth_start = NebulaPointCloud();
-    point_azimuth_end = NebulaPointCloud();
-    point_timestamp_start = NebulaPointCloud();
-    point_timestamp_end = NebulaPointCloud();
+    distance_filtered_count = 0;
+    fov_filtered_count = 0;
+    timestamp_filtered_count = 0;
+    invalid_point_count = 0;
+    multiple_return_point_count = 0;
+    mutliple_return_point_count = 0;
+    total_kept_point_count = 0;
+    invalid_packet_count = 0;
+    cloud_distance_min_m = 0;
+    cloud_distance_max_m = 0;
+    cloud_azimuth_min_rad = 0;
+    cloud_azimuth_max_rad = 0;
+    packet_timestamp_min_ns = 0;
+    packet_timestamp_max_ns = 0;
   }
 
   [[nodiscard]] nlohmann::ordered_json to_json() const
   {
-    nlohmann::ordered_json j;
-    j["distance_counter"] = distance_counter;
-    j["fov_counter"] = fov_counter;
-    j["timestamp_counter"] = timestamp_counter;
-    j["distance_start"] = distance_start;
-    j["distance_end"] = distance_end;
-    j["raw_azimuth_start"] = raw_azimuth_start;
-    j["raw_azimuth_end"] = raw_azimuth_end;
-    j["packet_timestamp_start"] = packet_timestamp_start;
-    j["packet_timestamp_end"] = packet_timestamp_end;
+    nlohmann::json distance_j;
+    distance_j["filter"] = "distance";
+    distance_j["distance_filtered_count"] = distance_filtered_count;
+    distance_j["cloud_distance_min_m"] = cloud_distance_min_m;
+    distance_j["cloud_distance_max_m"] = cloud_distance_max_m;
+    nlohmann::json fov_j;
+    fov_j["filter"] = "fov";
+    fov_j["fov_filtered_count"] = fov_filtered_count;
+    fov_j["cloud_azimuth_min_rad"] = cloud_azimuth_min_rad;
+    fov_j["cloud_azimuth_max_rad"] = cloud_azimuth_max_rad;
+    nlohmann::json timestamp_j;
+    timestamp_j["filter"] = "timestamp";
+    timestamp_j["timestamp_filtered_count"] = timestamp_filtered_count;
+    timestamp_j["packet_timestamp_min_ns"] = packet_timestamp_min_ns;
+    timestamp_j["packet_timestamp_max_ns"] = packet_timestamp_max_ns;
+    nlohmann::json invalid_j;
+    invalid_j["filter"] = "invalid";
+    invalid_j["invalid_point_count"] = invalid_point_count;
+    invalid_j["invalid_packet_count"] = invalid_packet_count;
+    nlohmann::json identical_j;
+    identical_j["filter"] = "identical";
+    identical_j["multiple_return_point_count"] = multiple_return_point_count;
+    nlohmann::json multiple_j;
+    multiple_j["filter"] = "multiple";
+    multiple_j["mutliple_return_point_count"] = mutliple_return_point_count;
```

Comment:
Please separate the measured pointcloud bounds (distance/azimuth/time) from the filter pipeline information.
Also make sure that the keys for each filter are called `"name"` and `"filtered_count"`, regardless of filter type:

```json
"filter_pipeline": [
  {"name": "distance", "filtered_count": 300},
  ...
],
"pointcloud_bounds": {
  "azimuth_deg": {
    "min": 0.0,
    "max": 270.0
  },
  // same with timestamp and distance
}
```

`invalid_packet_count` has very different causes than `invalid_point_count`, so I'd like to keep them separate.
Make it a top-level JSON property instead (same level as `total_kept_point_count`.